kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -633,10 +633,26 @@ printNotFound = putStrLn' "Variable or index not found"
 putStrLn' :: MonadWriter String m => String -> m ()
 putStrLn' str = tell $ str <> "\n"
 
-showDotGraph :: Graph gr => gr nl el -> IO ()
-showDotGraph =
+showDotGraph :: Int -> GraphType -> IO ()
+showDotGraph len =
     (flip Graph.runGraphvizCanvas') Graph.Xlib
-        . Graph.graphToDot Graph.nonClusteredParams
+        . Graph.graphToDot params
+  where
+    params = Graph.nonClusteredParams
+        { Graph.fmtEdge = \(_, _, l) ->
+            [Graph.textLabel (ruleIndex len l)]
+        }
+    ruleIndex lth lbl =
+        case listToMaybe . toList $ lbl of
+            Nothing -> "Simpl/RD"
+            Just rule -> LT.pack
+                      . (axiomOrClaim lth)
+                      . fromJust
```

Comment:
Rather than causing the repl to die here, I would probably prefer to `fmap axiomOrClaim lth` and do `maybe "Unknown" LT.pack`.